fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -143,14 +149,25 @@ where
     /// The results are never going to be flushed, so it's semantically read-only,
     /// but it might write into the buffered block store the FVM creates. Running
     /// multiple such messages results in their buffered effects stacking up.
-    pub fn call(&self, mut msg: FvmMessage) -> anyhow::Result<ApplyRet> {
+    pub fn call(&self, msg: FvmMessage) -> anyhow::Result<ApplyRet> {
+        // the default behavior of call is to reuse the cache and stack the results.
+        // for isolated execution of message without cache, use directly `call_with_cache`
+        self.call_with_cache(msg, true)
```

Comment:
So currently if one mixes `true` and `false` calls, then whether effects stack up depends on the order of execution, e.g. `call; call_with_cache(false); call` would see the cache reverted for the last call, while `call; call;` would all stack up. 

I would perhaps use a simpler semantic where the `reuse_cache: false` option would skip overwriting whatever was in the cache and leave it untouched, just working with a fresh engine, so the `true` and `false` values are completely isolated. I might be wrong on this, but it's always possible to go to an enum if one wants to specify the exact behaviour. 

I just find it odd that we have `call` which hides this from us. Maybe a `call` and a `call_nocache` with no flag could make it clear that one is stacking, the other isn't?

Reply:
We agreed to change this to have a single `call` method and add a `use_cache: bool` parameter to it. If `use_cache` is false it will not touch the cache, including not resetting it. This way it's always clear from the call site how it's going to work.